### PR TITLE
do not hard code the ubuntu distributor name

### PR DESCRIPTION
**Pull request by hchenxa**
**What this PR does / why we need it**:

the distributor name was hard code with `xenial`, prefer to not to hard code the distributor code name and get the name with command.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note

```

### CODE REVIEW

**Comment by hchenxa** [comment]
cc @gyliu513

File: cmd/clusterctl/examples/openstack/provider-component/user-data/ubuntu/templates/master-user-data.sh [hunk 1]
```diff
@@ -13,12 +13,14 @@ CLUSTER_DNS_DOMAIN={{ .Cluster.Spec.ClusterNetwork.ServiceDomain }}
...
 ARCH=amd64
+DISTRIBUTOR=$(lsb_release -cs)
+
 swapoff -a
 # disable swap in fstab
 sed -i.bak -r 's/(.+ swap .+)/#\1/' /etc/fstab
 curl -s https://packages.cloud.google.com/apt/doc/apt-key.gpg | sudo apt-key add -
 touch /etc/apt/sources.list.d/kubernetes.list
-sh -c 'echo "deb http://apt.kubernetes.io/ kubernetes-xenial main" > /etc/apt/sources.list.d/kubernetes.list'
+sh -c 'echo "deb http://apt.kubernetes.io/ kubernetes-${DISTRIBUTOR} main" > /etc/apt/sources.list.d/kubernetes.list'
```
Comment (hidekazuna) [inline cmd/clusterctl/examples/openstack/provider-component/user-data/ubuntu/templates/master-user-data.sh]: Ubuntu 18.04 lsb_release -cs returns bionic, but there is no kubernetes-bionic: https://packages.cloud.google.com/apt/dists

File: cmd/clusterctl/examples/openstack/provider-component/user-data/ubuntu/templates/worker-user-data.sh [hunk 2]
```diff
@@ -20,7 +21,7 @@ apt-get update
 apt-get install -y apt-transport-https prips
 apt-key adv --keyserver hkp://keyserver.ubuntu.com --recv-keys F76221572C52609D
 cat <<EOF > /etc/apt/sources.list.d/k8s.list
-deb [arch=amd64] https://apt.dockerproject.org/repo ubuntu-xenial main
+deb [arch=amd64] https://apt.dockerproject.org/repo ubuntu-${DISTRIBUTOR} main
```
Comment (hidekazuna) [inline cmd/clusterctl/examples/openstack/provider-component/user-data/ubuntu/templates/worker-user-data.sh]: Ubuntu 18.04 lsb_release -cs returns bionic, but there is no kubernetes-bionic: https://apt.dockerproject.org/repo/dists/

Comment (jichenjc) [reply cmd/clusterctl/examples/openstack/provider-component/user-data/ubuntu/templates/worker-user-data.sh]: you are right, for bionic I tested and it works with ubuntu-xenial 

so seems this should not be a good PR now, thanks for bring this out @hidekazuna

**Comment by sbueringer** [comment]
@jichenjc I think that's also not our job anymore :). Should be done differently either way in CAPI as a whole. I think the kubeadm bootstrapper relies on images where dependencies are already installed before cloud-init starts

**Comment by sbueringer** [comment]
I'll close this PR because it's outdated. We do not provide the user data anymore since v1beta2.

**Pull request closed without merging**